### DISCUSSION TITLE
Allow conditions on directus_users

### DISCUSSION THREAD

**Discussion by Attacler** [directus/directus#10326] [Feature Requests]
Hi there,

For some projects not every field is necessary to be shown on the user.
Is it somehow possible to apply conditions to it?

**Comment by Attacler** [comment]
Maybe its better to give the user full controll over the frontend so the user/admin has full controll over it

**Comment by rijkvanzanten** [comment]
System collections can't be changed, as the configuration for the fields / rows isn't stored in the database.

In order to support this, we'd have to come up with some smart way to merge any changes made in the DB directly on top of the default row that's provided by the API.

**Reply by Attacler** [reply]
> System collections can't be changed, as the configuration for the fields / rows isn't stored in the database.
> 
> In order to support this, we'd have to come up with some smart way to merge any changes made in the DB directly on top of the default row that's provided by the API.

True but basicly what i was thinking about the following things to be allowed for an admin to change:
![image](https://user-images.githubusercontent.com/42644889/145175886-bf3e80a4-b3c8-497b-afa7-c48eb4367d4f.png)
![image](https://user-images.githubusercontent.com/42644889/145175930-a922e684-1087-4014-a60a-192677c63506.png)
This way the database stays the same.
But maybe something needs to be changed with migrations or something.

I would prefer to make the conditions work, this would allow you to make the following example:
admin users are allowed to have a token, so every role that is differend, would not show this field.
same counts for thema and translation.

**Reply by rijkvanzanten** [reply]
Yeah, what I meant by _System collections can't be changed, as the configuration for the fields / rows isn't stored in the database._ is that that's currently not technically possible without some new magic, as the configuration you highlighted in the screenshots isn't stored in the database itself, so can't be altered on the fly

**Reply by Attacler** [reply]
Ah right, i see now.
I guess i`ll stick to custom css for now.
If anyone wants to pick this up, ill throw in a 20€ trough issuehunt.

**Reply by Attacler** [reply]
> Yeah, what I meant by _System collections can't be changed, as the configuration for the fields / rows isn't stored in the database._ is that that's currently not technically possible without some new magic, as the configuration you highlighted in the screenshots isn't stored in the database itself, so can't be altered on the fly

Could you give me an estimate of the costs of someone changing this for all system collections? (I`ll discuss with the co owners of my company if its worth it based on this)

**Reply by benhaynes** [reply]
Perhaps @rijkvanzanten can provide a rough idea of timing/cost (here or in Discord).

**Reply by rijkvanzanten** [reply]
Hmm, we'd have to implement some custom checks on the /fields endpoint that will merge any custom configuration stored in the database on top of our defaults. We'll also have to make sure we limit what things you can alter, as you still won't be able to delete fields for example 🤔

**Reply by Attacler** [reply]
> Hmm, we'd have to implement some custom checks on the /fields endpoint that will merge any custom configuration stored in the database on top of our defaults. We'll also have to make sure we limit what things you can alter, as you still won't be able to delete fields for example 🤔

I suggest that you can only change the frontend/directusapp side and not the actual datastructure.
Maybe do show the datastructure but with a notice and readonly.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
